src/components/va-radio/va-radio.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Listen,
+  Prop,
+  h,
+} from '@stencil/core';
+import { getSlottedNodes } from '../../utils/utils';
+
+@Component({
+  tag: 'va-radio',
+  styleUrl: 'va-radio.css',
+  shadow: true,
+})
+export class VaRadio {
+  @Element() el: any;
+
+  @Prop() label: string;
+
+  @Prop() required: boolean = false;
+
+  @Prop() error: string;
+
+  @Prop() enableAnalytics: boolean;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaValueChange: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaKeyDown: EventEmitter;
+
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  vaMouseDown: EventEmitter;
+
+  @Listen('radioOptionSelected')
+  radioOptionSelectedHandler(event: CustomEvent): void {
+    const clickedItem = event.target as HTMLVaRadioOptionElement;
+
+    getSlottedNodes(this.el, 'va-radio-option')
+      .filter(item => item !== clickedItem)
+      .forEach(item => ((item as HTMLVaRadioOptionElement).checked = false));
```

Comment:
:thinking: Is there a simple way to either ensure each slotted node is a radio button _or_ filter the list of slotted nodes by type to only grab the radio buttons so we can be sure each item _is_ in fact an `HTMLVaRadioOptionElement`?

Reply:
I don't think we can put limits on which slotted nodes are _allowed_. The closest thing might be to have a `::slotted(*) { display: none;}` followed up by `::slotted(va-radio-option) { display: block;}`, but that's just for presentation.

Regarding the filtering by type, we might be able to do that here if I'm understanding what you're asking:

https://github.com/department-of-veterans-affairs/component-library/blob/4ee6bb1b0dfe33f029d13b8ec5cd10b61bffdcb3/src/utils/utils.ts#L27-L29

I'll see what a refactor looks like based on type as opposed to the node name.

Reply:
:man_facepalming: I think what I'm actually asking is for `getSlottedNodes` to do what it already does. I totally didn't read the part where this _is_ getting all `va-radio-option`s. That's my bad. You're good here. :sweat_smile: